src/filters/GeltungsbereicheMenuFilter/GeltungsbereicheMenuFilter.js

```diff
@@ -0,0 +1,167 @@
+import React, { useState, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import {
+  InputLabel,
+  MenuItem,
+  IconButton,
+  makeStyles,
+} from '@material-ui/core';
+import { MdClear } from 'react-icons/md';
+import Select from '../../components/Select';
+import geltungsbereicheMapping from '../../utils/geltungsbereicheMapping.json';
+
+const useStyles = makeStyles(() => ({
+  selectWrapper: {
+    position: 'relative',
+  },
+  paper: {
+    overflow: 'hidden auto',
+    maxHeight: 350,
+    width: 268,
+  },
+  item: {
+    display: 'block',
+    textOverflow: 'ellipsis',
+  },
+  input: {
+    width: 270,
+  },
+  value: {
+    maxWidth: 190,
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  clearBtn: {
+    position: 'absolute',
+    top: 14,
+    right: (props) => (props.isMobile ? 40 : 30),
+  },
+}));
+
+const propTypes = {
+  topic: PropTypes.shape().isRequired,
+};
+
+const GeltungsbereicheMenuFilter = ({ topic }) => {
+  const { t, i18n } = useTranslation();
+  const nullOption = {
+    value: 'gb-null',
+    label: 'Alle Produkte',
+  };
+  const [filterValue, setFilterValue] = useState(nullOption);
+  const screenWidth = useSelector((state) => state.app.screenWidth);
+  const isMobile = useMemo(() => {
+    return ['xs'].includes(screenWidth);
+  }, [screenWidth]);
+  const classes = useStyles({ isMobile });
+
+  const providerChoices = useMemo(
+    () => [
+      nullOption,
+      ...Object.keys(geltungsbereicheMapping).map((key) => {
+        return {
+          value: key,
+          label: geltungsbereicheMapping[key],
+        };
+      }),
+    ],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [i18n.language],
+  );
+
+  const layers = topic.layers.filter((l) => {
+    return /^ch.sbb.geltungsbereiche-/.test(l.key);
+  });
+
+  const onChange = (value) => {
+    layers.flat().forEach((layer) => {
+      if (layer) {
+        const { mbMap } = layer.mapboxLayer;
+        const style = mbMap.getStyle();
+        if (!mbMap || !style) {
+          return;
+        }
+        const styleLayer = style.layers.find((l) => {
```

Comment:
Why don't you use layer.styleLayersFilter to find the corresponding layers in the style ?